src/saveload/airport_sl.cpp

```diff
@@ -14,29 +14,39 @@
 
 #include "../safeguards.h"
 
-static void Save_APID()
-{
-	Save_NewGRFMapping(_airport_mngr);
-}
-
-static void Load_APID()
-{
-	Load_NewGRFMapping(_airport_mngr);
-}
-
-static void Save_ATID()
-{
-	Save_NewGRFMapping(_airporttile_mngr);
-}
-
-static void Load_ATID()
-{
-	Load_NewGRFMapping(_airporttile_mngr);
-}
-
-static const ChunkHandler airport_chunk_handlers[] = {
-	{ 'ATID', Save_ATID, Load_ATID, nullptr, nullptr, CH_TABLE },
-	{ 'APID', Save_APID, Load_APID, nullptr, nullptr, CH_TABLE },
+struct APIDChunkHandler : ChunkHandler {
+	APIDChunkHandler() : ChunkHandler('APID', CH_TABLE) {}
+
+	void Save() const override
+	{
+		Save_NewGRFMapping(_airport_mngr);
```

Comment:
Might not be really in scope here, but all these NewGRF mappings are essentially the same except for the ID and the reference table. May it be useful to just have one NewGRFMappingChunkHandler with chunk ID and a reference to the table?

Reply:
Ah yes, that's one of the thing this PR allows.